ml-drift: handle 5D Reshape3DAttributes/Transpose3DAttributes

Adds std::any_cast branches for Reshape3DAttributes and
Transpose3DAttributes in the legacy GPUOperationFromNode selector.

diff --git a/ml_drift/common/selectors/operation_selector.cc b/ml_drift/common/selectors/operation_selector.cc
index a729d68..60302f2 100644
--- a/ml_drift/common/selectors/operation_selector.cc
+++ b/ml_drift/common/selectors/operation_selector.cc
@@ -1022,11 +1029,17 @@ absl::Status GPUOperationFromNode(const GpuInfo& gpu_info,
       return absl::OkStatus();
     }
     case OperationType::RESHAPE: {
-      const auto& attr =
-          std::any_cast<const ReshapeAttributes&>(node.operation.attributes);
       ABSL_ASSIGN_OR_RETURN(auto src, model_builder->GetTensor(inputs[0]->id));
-      auto output = model_builder->Reshape(src, attr.new_shape);
-      return model_builder->UpdateOutputTensor(output, outputs[0]->id);
+      if (auto* attr3d = std::any_cast<Reshape3DAttributes>(
+              &node.operation.attributes)) {
+        auto output = model_builder->Reshape(src, attr3d->new_shape);
+        return model_builder->UpdateOutputTensor(output, outputs[0]->id);
+      } else {
+        const auto& attr =
+            std::any_cast<const ReshapeAttributes&>(node.operation.attributes);
+        auto output = model_builder->Reshape(src, attr.new_shape);
+        return model_builder->UpdateOutputTensor(output, outputs[0]->id);
+      }
     }
     case OperationType::RESIZE: {
       const auto& attr =
@@ -1122,11 +1135,17 @@ absl::Status GPUOperationFromNode(const GpuInfo& gpu_info,
           model_builder->TopK(src, attr.k), {outputs[0]->id, outputs[1]->id});
     }
     case OperationType::TRANSPOSE: {
-      const auto& attr =
-          std::any_cast<const TransposeAttributes&>(node.operation.attributes);
       ABSL_ASSIGN_OR_RETURN(auto src, model_builder->GetTensor(inputs[0]->id));
-      return model_builder->UpdateOutputTensor(
-          model_builder->Transpose(src, attr.perm), outputs[0]->id);
+      if (auto* attr3d = std::any_cast<Transpose3DAttributes>(
+              &node.operation.attributes)) {
+        return model_builder->UpdateOutputTensor(
+            model_builder->Transpose(src, attr3d->perm), outputs[0]->id);
+      } else {
+        const auto& attr =
+            std::any_cast<const TransposeAttributes&>(node.operation.attributes);
+        return model_builder->UpdateOutputTensor(
+            model_builder->Transpose(src, attr.perm), outputs[0]->id);
+      }
     }
     default:
       ABSL_ASSIGN_OR_RETURN(auto gpu_op, SelectDefault(gpu_info, op_def, node));
